Jenkinsfile

```diff
@@ -11,109 +11,192 @@ pipeline {
+        POMFIX = 'libs/libs-pomfix,libs/libs-pomfix-run'
+        LARGE_MVN_OPTS = '-Xmx8192M -Xss128M -XX:+CMSClassUnloadingEnabled -XX:+UseConcMarkSweepGC '
+        LINUX_MVN_RANDOM = '-Djava.security.egd=file:/dev/./urandom'
     }
     stages {
         stage('Setup') {
             steps {
                 slackSend color: 'good', message: "STARTED: ${JOB_NAME} ${BUILD_NUMBER} ${BUILD_URL}"
             }
         }
-        stage('Parallel Build') {
+        // Use the pomfix tool to validate that bundle dependencies are properly declared
+        stage('Validate Poms') {
+            agent { label 'linux-small' }
+            steps {
+                retry(3) {
+                    checkout scm
+                }
+                withMaven(maven: 'M3', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LINUX_MVN_RANDOM}') {
+                    sh 'mvn clean install -DskipStatic=true -DskipTests=true -pl $POMFIX'
+                }
+            }
+        }
+        // The incremental build will be triggered only for PRs. It will build the differences between the PR and the target branch
+        stage('Incremental Build') {
+            when {
+                allOf {
+                    expression { env.CHANGE_ID != null }
+                    expression { env.CHANGE_TARGET != null }
+                }
+            }
             // TODO CAL-296 refactor this stage from scripted syntax to declarative syntax to match the rest of the stages - https://issues.jenkins-ci.org/browse/JENKINS-41334
             steps {
                 parallel(
-                    linux: {
-                        node('linux-large') {
-                            retry(3) {
-                                checkout scm
+                        linux: {
+                            node('linux-large') {
+                                retry(3) {
+                                    checkout scm
+                                }
+                                timeout(time: 1, unit: 'HOURS') {
+                                    withMaven(maven: 'M35', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS} ${LINUX_MVN_RANDOM}', options: [artifactsPublisher(disabled: true), dependenciesFingerprintPublisher(disabled: true, includeScopeCompile: false, includeScopeProvided: false, includeScopeRuntime: false, includeSnapshotVersions: false)]) {
+                                        sh 'mvn install -pl !$DOCS -DskipStatic=true -DskipTests=true -T 1C'
+                                        sh 'mvn clean install -B -T 1C -pl !$ITESTS -Dgib.enabled=true -Dgib.referenceBranch=/refs/remotes/origin/$CHANGE_TARGET'
+                                        sh 'mvn install -B -Dmaven.test.redirectTestOutputToFile=true -pl $ITESTS -nsu'
```

Comment:
perhaps we should remove `-Dmaven.test.redirectTestOutputToFile=true`

Reply:
will do